### PR TITLE
[Symfony] AddFlashRector

### PR DESCRIPTION
**Pull request by mssimi**
Hey added rector simplifying adding flash messages. Helper method addFlash was introduced in SF 2.6.

```php
/**
 * Before:
 * $request->getSession()->getFlashBag()->add('success', 'something');
 * After:
 * $this->addflash('success', 'something');
 */
```
It is not finished, if you could help me out, It will be great. I added tests, config and I need help with finishing rector as i'm stuck with it now. Thx.

### CODE REVIEW

File: src/Rector/Contrib/Symfony/Controller/AddFlashRector.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+use Rector\Node\Attribute;
+use Rector\Node\MethodCallNodeFactory;
+use Rector\NodeAnalyzer\MethodCallAnalyzer;
+use Rector\Rector\AbstractRector;
+
+/**
+ * Before:
+ * $request->getSession()->getFlashBag()->add('success', 'something');
+ * After:
+ * $this->>addflash('success', 'something');
```
Comment (carusogabriel) [inline src/Rector/Contrib/Symfony/Controller/AddFlashRector.php]: Double `->>`

File: tests/Rector/Contrib/Symfony/Controller/AddFlashRector/AddFlashRectorTest.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Tests\Rector\Contrib\Symfony\Controller\RedirectToRouteRector;
```
Comment (carusogabriel) [inline tests/Rector/Contrib/Symfony/Controller/AddFlashRector/AddFlashRectorTest.php]: Namespace wrong. Should be `Rector\Tests\Rector\Contrib\Symfony\Controller\AddFlashRector`

File: tests/Rector/Contrib/Symfony/Controller/AddFlashRector/Correct/correct2.php.inc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php declare (strict_types=1);
+
+use Symfony\Component\HttpFoundation\Request;
+
+class AnyClass
+{
+    public function someAction(Request $request)
+    {
+        $request->getSession()->getFlashBag()->add('success', 'message');
```
Comment (carusogabriel) [inline tests/Rector/Contrib/Symfony/Controller/AddFlashRector/Correct/correct2.php.inc]: Shouldn't it be `$this->addflash('success', 'something');` as is a correct file?

Comment (mssimi) [reply tests/Rector/Contrib/Symfony/Controller/AddFlashRector/Correct/correct2.php.inc]: No, its not a Controller.

**Comment by TomasVotruba** [comment]
How do you like this?
https://github.com/rectorphp/rector/pull/312

Feel free to cherry pick it or copy paste it.

**Comment by mssimi** [comment]
I think we got it:) Thx for help, that method is exactly what I needed and I think it is very useful method for future rectors:)

**Comment by TomasVotruba** [comment]
Thanks for your work :+1:

**Pull request merged**